akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Monitor.java

```diff
@@ -0,0 +1,84 @@
+package jdocs.stream.operators.sourceorflow;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.japi.Pair;
+import akka.japi.pf.Match;
+import akka.stream.FlowMonitor;
+import akka.stream.FlowMonitorState;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import scala.PartialFunction;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ *
+ */
+public class Monitor {
+
+    private static final PartialFunction<Object, NotUsed> printMonitorState = Match.match(
+            FlowMonitorState.Initialized$.class,
+            __ -> {
+                System.out.println("Stream is initialized but hasn't processed any element");
+                return NotUsed.getInstance();
+            }
+    ).match(
+            FlowMonitorState.Received.class,
+            (msg) -> {
+                System.out.println("Last message received: " + msg.msg());
+                return NotUsed.getInstance();
+            }
+    ).match(
+            FlowMonitorState.Failed.class,
+            (failed) -> {
+                System.out.println("Stream failed with cause: " + failed.cause().getMessage());
+                return NotUsed.getInstance();
+            }
+    ).match(
+            FlowMonitorState.Finished$.class,
+            __ -> {
+                System.out.println("Stream completed already");
+                return NotUsed.getInstance();
+            }
+    ).build();
```

Comment:
I don't think we should expose Java users to the PF if we don't explicitly need to pass a PF somewhere, i think regular method with instanceof is what Java devs would expect/understand right away.